backend/llm_integration.py

```
@@ -342,15 +342,62 @@ def get_embedding_client(provider_type: str, model_name: str = None, api_key: st
+    except (ValueError, TypeError):
+        return None
+    if normalized.startswith(_allowed_model_roots()):
+        return normalized
+    logger.warning(
+        "Configured model path is outside the allowed roots "
+        "(models/, home directory, DOCU_MODEL_ROOTS): %s", str(model_path).replace('\n', '\\n')
+    )
+    return None
+
+
 def get_local_llm(model_path: str, tensor_split: List[float] = None) -> Any:
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)